midi_app_controller/gui/_tests/test_binds_editor.py

```diff
@@ -0,0 +1,328 @@
+    knob_binds_fixture = request.getfixturevalue(knob_binds)
+    assert len(knob_binds_fixture.knob_combos) == expected_length
+
+
+@pytest.mark.parametrize(
+    "button_binds, expected_binds",
+    [
+        (
+            "knob_binds_fixture",
+            [
+                ("volume_up", "volume_down"),
+                ("zoom_in", "zoom_out"),
+            ],
+        ),
+        (
+            "knob_binds_fixture_mixed",
+            [
+                ("volume_up", "volume_down"),
+            ],
+        ),
+    ],
+)
+def test_get_knob_binds(request, button_binds, expected_binds):
+    fixture = request.getfixturevalue(button_binds)
+    binds = fixture.get_binds()
+    assert len(binds) == len(expected_binds)
+    for bind, (expected_increase, expected_decrease) in zip(binds, expected_binds):
+        assert bind.action_id_increase == expected_increase
+        assert bind.action_id_decrease == expected_decrease
+
+
+def test_binds_editor_switch_views(binds_editor_fixture_basic):
+    widget = binds_editor_fixture_basic
+    QTest.mouseClick(widget.buttons_radio, Qt.LeftButton)
+    QTest.qWait(100)
+    assert widget.buttons_widget.isVisible()
+    assert not widget.knobs_widget.isVisible()
+
+    QTest.mouseClick(widget.knobs_radio, Qt.LeftButton)
+    QTest.qWait(100)
+    assert widget.knobs_widget.isVisible()
+    assert not widget.buttons_widget.isVisible()
+
+
+def test_binds_editor_save_and_exit(binds_editor_fixture):
+    widget, mock_save_and_exit, mock_exit = binds_editor_fixture
+    save_and_exit_button = widget.save_and_exit_button
+    assert save_and_exit_button is not None
+
+    QTest.mouseClick(save_and_exit_button, Qt.LeftButton)
+    QTest.qWait(100)
+
+    mock_save_and_exit.assert_called_once()
+    mock_exit.assert_not_called()
+
+
+def test_binds_editor_exit(binds_editor_fixture):
+    widget, mock_save_and_exit, mock_exit = binds_editor_fixture
+    exit_button = widget.exit_button
+    assert exit_button is not None
+
+    QTest.mouseClick(exit_button, Qt.LeftButton)
+    QTest.qWait(100)
+
+    mock_exit.assert_called_once()
+    mock_save_and_exit.assert_not_called()
+
+
+def test_binds_editor_save_and_exit_no_mock(binds_editor_fixture_basic):
+    widget = binds_editor_fixture_basic
+    save_and_exit_button = widget.save_and_exit_button
+    assert save_and_exit_button is not None
+
+    QTest.mouseClick(save_and_exit_button, Qt.LeftButton)
+    QTest.qWait(100)
+
+    assert not widget.isVisible()
+
+
+def test_binds_editor_exit_no_mock(binds_editor_fixture_basic):
+    widget = binds_editor_fixture_basic
+    exit_button = widget.exit_button
+    assert exit_button is not None
+
+    QTest.mouseClick(exit_button, Qt.LeftButton)
+    QTest.qWait(100)
+
+    assert not widget.isVisible()
+
+
+def test_button_binds_initialization_unbound(qtbot, controller_sample):
+    unbound_buttons = [ControllerElement(id=3, name="Unbound Button")]
+    actions = ["play_action", "stop_action"]
```

Comment:
```suggestion
    actions_str = ["play_action", "stop_action"]
```
Maybe change the variable name so it doesn't change its type in the line below. The current way is fine, but this looks cleaner to me.